lib-cloudsdk/src/main/java/com/facebook/openwifi/cloudsdk/AggregatedState.java

```diff
@@ -109,64 +65,161 @@ public boolean equals(Object obj) {
+	 */
+	public static class AssociationInfo {
+		/** Rate information with aggregated fields. */
+		public static class Rate {
+			/**
+			 * Aggregated fields bitRate
+			 */
+			public long bitRate;
+
+			/**
+			 * Aggregated fields chWidth
+			 */
+			public int chWidth;
+
+			/**
+			 * Aggregated fields mcs
+			 */
+			public int mcs;
+
+			/** Constructor with no args */
+			private Rate() {}
+
+			/** Constructor with args */
+			private Rate(long bitRate, int chWidth, int mcs) {
+				this.bitRate = bitRate;
+				this.chWidth = chWidth;
+				this.mcs = mcs;
+			}
+		}
+
+		public long connected;
+		public long inactive;
+		public int rssi;
+		public long rxBytes;
+		public long rxPackets;
+		public Rate rxRate;
+		public long txBytes;
+		public long txDuration;
+		public long txFailed;
+		public long txPackets;
+		public Rate txRate;
+		public long txRetries;
+		public int ackSignal;
+		public int ackSignalAvg;
+		public long txPacketsCounters;
+		public long txErrorsCounters;
+		public long txDroppedCounters;
+		public long activeMsRadio;
+		public long busyMsRadio;
+		public long noiseRadio;
+		public long receiveMsRadio;
+		public long transmitMsRadio;
+
+		/** unix time in ms */
+		public long timestamp;
+
+		/** Default Constructor. */
+		public AssociationInfo() {}
+
+		/** Constructor with only rssi(for test purpose) */
+		public AssociationInfo(int rssi) {
+			this.rssi = rssi;
+		}
+
+		/** Constructor with args */
+		public AssociationInfo(
+			Association association,
+			Counters counters,
+			JsonObject radio,
+			long timestamp
+		) {
+			// Association info
+			connected = association.connected;
+			inactive = association.inactive;
+			rssi = association.rssi;
+			rxBytes = association.rx_bytes;
+			rxPackets = association.rx_packets;
+			if (association.rx_rate != null) {
+				rxRate = new Rate(
+					association.rx_rate.bitrate,
+					association.rx_rate.chwidth,
+					association.rx_rate.mcs
+				);
+			} else {
+				rxRate = new Rate();
+			}
+
+			txBytes = association.tx_bytes;
+			txPackets = association.tx_packets;
+
+			if (association.tx_rate != null) {
+				txRate = new Rate(
+					association.tx_rate.bitrate,
+					association.tx_rate.chwidth,
+					association.tx_rate.mcs
+				);
+			} else {
+				txRate = new Rate();
+			}
+			txRetries = association.tx_retries;
+			ackSignal = association.ack_signal;
+			ackSignalAvg = association.ack_signal_avg;
+
+			//Counters info
+			txPacketsCounters = counters.tx_packets;
+			txErrorsCounters = counters.tx_errors;
+			txDroppedCounters = counters.tx_dropped;
+
+			// Radio info
+			activeMsRadio = radio.get("active_ms").getAsLong();
+			busyMsRadio = radio.get("busy_ms").getAsLong();
+			transmitMsRadio = radio.get("transmit_ms").getAsLong();
+			receiveMsRadio = radio.get("receive_ms").getAsLong();
+			noiseRadio = radio.get("noise").getAsLong();
+
+			this.timestamp = timestamp;
+		}
+	}
+
+	/** Aggregate AssociationInfo over bssid, station and RadioConfig */
```

Comment:
don't put javadoc comments when they aren't tied to a specific item - use a regular comment